addons/arrays/fnc_selectRandom.sqf

```diff
@@ -0,0 +1,37 @@
+#include "script_component.hpp"
+/* ----------------------------------------------------------------------------
+Function: CBA_fnc_selectRandom
+
+Description:
+    Select a specified amount of elements from an array without picking the same element multiple times.
+
+Parameters:
+    _array - Input Array <ARRAY>
+    _amount - Block <NUMBER>
+
+Returns:
+    New array with the specified amount of randomly selected elements <ARRAY>
+
+Example:
+    (begin example)
+        _result = [[1, 2, 3, 4, 5], 2] call CBA_fnc_selectRandom;
+        // _result => [2, 4] (random)
+    (end)
+
+Author:
+    NeilZar
+---------------------------------------------------------------------------- */
+SCRIPT(selectRandom);
+
+params [["_array", [], [[]]], ["_amount", 0, [0]]];
+
+if (_amount >= count _array) exitWith { _array };
+if (_amount <= 0) exitWith { [] };
```

Comment:
This line is superfluous. 

Suppose it were deleted and `_amount` would be less or equal than zero.
```sqf
//if (_amount <= 0) exitWith { [] };

private _result = [];

for "_i" from 0 to _amount - 1 do {
    _result pushBack (_array deleteAt floor random count _array)
};

_result
```
Then `_amount - 1` would always be less than zero.
A for loop from `A` to `B` where `B` < `A` would just be ignored. The code block would never be executed.
This means that `_result` would be set to `[]` and never be modified. The return value would always be `[]`, so the same as the return value of the `exitWith` block.

By deleting this line, you also remove one additional exit point from the function, and having as few of these as possible (preferably none) is generally a sign of good clean code.